lms/static/js/ccx/schedule.js

```diff
@@ -251,9 +264,27 @@ var edx = edx || {};
 	    }
 	},
 
+	valid_dates: function(start, due) {
+          if (!due) {
+            return [true, "ok"];
+          } else if (!start && due) {
+            return [false, "Due date cannot be before start date." +
+              " Please enter valid start date and time"];
+          }
+
+          var dueDate = new Date(due);
+          var startDate = new Date(start);
+
+          if (dueDate < startDate) {
+            return [false, "Due date cannot be before start date."];
+          }
+          return [true, "ok"];
+	},
```

Comment:
why you don't just return a list of errors? if the `list.length` is `0`, the date is valid (so on line 119 it's easier to understand what's going on)